src/olympia/accounts/tests/test_views.py

```diff
@@ -497,19 +497,49 @@ def test_code_not_provided(self):
         assert not self.find_user.called
         assert not self.fxa_identify.called
 
-    def test_logged_in_disallows_login(self):
+    @mock.patch.object(views, 'generate_api_token')
+    def test_logged_in_disallows_login(self, generate_api_token_mock):
+        self.request.data = {
+            'code': 'foo',
+            'state': 'some-blob:{}'.format(base64.urlsafe_b64encode('/next')),
+        }
+        self.user = UserProfile()
+        self.request.user = self.user
+        assert self.user.is_authenticated()
+        self.request.COOKIES = {views.API_TOKEN_COOKIE: 'foobar'}
+        self.fn(self.request)
+        self.render_error.assert_called_with(
+            self.request, views.ERROR_AUTHENTICATED, next_path='/next',
+            format='json')
+        assert not self.find_user.called
+        assert self.render_error.return_value.set_cookie.call_count == 0
+        assert generate_api_token_mock.call_count == 0
+
+    @mock.patch.object(views, 'generate_api_token', lambda u: 'fake-api-token')
+    def test_already_logged_in_add_api_token_cookie_if_missing(self):
         self.request.data = {
             'code': 'foo',
             'state': 'some-blob:{}'.format(base64.urlsafe_b64encode('/next')),
         }
         self.user = UserProfile()
         self.request.user = self.user
         assert self.user.is_authenticated()
+        self.request.COOKIES = {}
         self.fn(self.request)
         self.render_error.assert_called_with(
             self.request, views.ERROR_AUTHENTICATED, next_path='/next',
             format='json')
         assert not self.find_user.called
+        response = self.render_error.return_value
+        assert response.set_cookie.call_count == 1
+        response.set_cookie.assert_called_with(
+            # Fatal flow: though we are able to set the max age of the cookie,
```

Comment:
Yeah, but I should remove that comment, it was from when I was trying to copy the expiration date...